### DISCUSSION TITLE
External YouTube Music Links to redirect to the app

### DISCUSSION THREAD

**Discussion by touzel** [pear-devs/pear-desktop#240] [Ideas]
Right now, when you search for a song on a search engine and click the link for YouTube Music, or a fellow user shares a link to you, it opens the browser non-premium ad-inclusive version of YouTube music. It would be amazing if a feature would enable it for this build to recognize YTMusic links and open them in it. If I'm missing something there is already such a feature, please point me towards it.

**Comment by Araxeus** [comment]
Could make a custom URL scheme in app (Protocol handler)
And then also make a chrome extension to detect youtube music links and launch said URL protocol

I'm just throwing ideas, not saying I'm doing it personally
Seems like alot of work to do *alone*, without that much benefits
But i'm willing to try and help out if anyone wants to work on that

> i'm obviously not the project owner, he might have more to say 😛

**Comment by naythanN** [comment]
I think it would be an amazing feature too.

**Comment by konhi** [comment]
I think [FreeTube](https://github.com/FreeTubeApp/FreeTube) has this feature. Might be useful for code reference.

**Comment by babelaids** [comment]
The project [ytmdesktop](https://github.com/ytmdesktop/ytmdesktop) does this, but unfortunately it doesn't have much contextual difference between YTM and youtube, so any time you click on any youtube link the app pops up and says "do you want to open this here?" Which 70% of the time I don't. That's why I switched to this app, but that feature could be tweaked for this. Point is it's possible
